tests/python/pants_test/backend/codegen/tasks/test_protobuf_integration.py

```diff
@@ -15,6 +15,12 @@
 
 class ProtobufIntegrationTest(PantsRunIntegrationTest):
 
+  def test_import_from_buildroot(self):
+    pants_run = self.run_pants(
+      ['gen.protoc', '--import-from-root',
+       'testprojects/src/protobuf/org/pantsbuild/testproject/import_from_buildroot/bar'])
+    self.assert_success(pants_run)
+
```

Comment:
I see what you mean. We've run into the source root import 'feature' a few times too. Another approach to resolving it is working towards isolation based on the build graph. That also won't require specific per task error handling, which is good.

I just get a bit nervous when I see a test suite that doesn't have any tests for failure / error cases.